src/Tizen.Multimedia.Remoting/MediaController/EnumExtensions.cs

```diff
@@ -21,79 +21,105 @@ namespace Tizen.Multimedia.Remoting
+                case MediaControlPlaybackCommand.Pause: return MediaControllerNativePlaybackAction.Pause;
+                case MediaControlPlaybackCommand.Stop: return MediaControllerNativePlaybackAction.Stop;
+                case MediaControlPlaybackCommand.Next: return MediaControllerNativePlaybackAction.Next;
+                case MediaControlPlaybackCommand.Previous: return MediaControllerNativePlaybackAction.Prev;
+                case MediaControlPlaybackCommand.FastForward: return MediaControllerNativePlaybackAction.FastForward;
+                case MediaControlPlaybackCommand.Rewind: return MediaControllerNativePlaybackAction.Rewind;
+                case MediaControlPlaybackCommand.Toggle: return MediaControllerNativePlaybackAction.Toggle;
             }
-            return MediaControllerPlaybackCode.Play;
+            return MediaControllerNativePlaybackAction.Play;
         }
 
-        internal static NativeRepeatMode ToNative(this MediaControlRepeatMode mode)
+        internal static MediaControllerNativeRepeatMode ToNative(this MediaControlRepeatMode mode)
         {
             Debug.Assert(Enum.IsDefined(typeof(MediaControlRepeatMode), mode));
 
-            return mode == MediaControlRepeatMode.Off ? NativeRepeatMode.On : NativeRepeatMode.Off;
+            return mode == MediaControlRepeatMode.Off ? MediaControllerNativeRepeatMode.On :
+                (mode == MediaControlRepeatMode.On ? MediaControllerNativeRepeatMode.Off : MediaControllerNativeRepeatMode.OneMedia);
         }
 
-        internal static MediaControlRepeatMode ToPublic(this NativeRepeatMode mode)
+        internal static MediaControlRepeatMode ToPublic(this MediaControllerNativeRepeatMode mode)
         {
-            Debug.Assert(Enum.IsDefined(typeof(NativeRepeatMode), mode));
+            Debug.Assert(Enum.IsDefined(typeof(MediaControllerNativeRepeatMode), mode));
 
-            return mode == NativeRepeatMode.Off ? MediaControlRepeatMode.On : MediaControlRepeatMode.Off;
+            return mode == MediaControllerNativeRepeatMode.Off ? MediaControlRepeatMode.On :
+                (mode == MediaControllerNativeRepeatMode.On ? MediaControlRepeatMode.Off : MediaControlRepeatMode.OneMedia);
+        }
+
+        internal static MediaControllerNativeShuffleMode ToNative(this MediaControlShuffleMode mode)
+        {
+            Debug.Assert(Enum.IsDefined(typeof(MediaControlShuffleMode), mode));
+
+            return mode == MediaControlShuffleMode.Off ? MediaControllerNativeShuffleMode.On : MediaControllerNativeShuffleMode.Off;
+        }
+
+        internal static MediaControlShuffleMode ToPublic(this MediaControllerNativeShuffleMode mode)
+        {
+            Debug.Assert(Enum.IsDefined(typeof(MediaControllerNativeShuffleMode), mode));
+
+            return mode == MediaControllerNativeShuffleMode.Off ? MediaControlShuffleMode.On : MediaControlShuffleMode.Off;
```

Comment:
It should be reversed

Reply:
ShuffleMode is added in the same order with RepeaeMode that announced in previous version.
So it can not be reversed.